pkg/recipes/terraform/config/config.go

```diff
@@ -150,13 +167,24 @@ func getProviderConfigs(ctx context.Context, requiredProviders []string, support
 			return nil, err
 		}
 		if len(config) > 0 {
-			providerConfigs[provider] = config
+			providerConfigs[provider] = []any{config}
 		}
 	}
 
 	return providerConfigs, nil
 }
 
+// getRecipeProviderConfigs returns the Terraform provider configurations for Terraform providers.
```

Comment:
Can you elaborate this comment to include context on where these configurations are coming from?